datacompy/core.py

```diff
@@ -275,11 +286,12 @@ def _dataframe_merge(self, ignore_spaces):
         # Clean up temp columns for duplicate row matching
         if self._any_dupes:
             if self.on_index:
-                outer_join.index = outer_join[index_column]
+                # outer_join.index = outer_join[index_column]
```

Comment:
Can lines 289-291 be further simplified to `outer_join.set_index(index_column, drop=True, inplace=True)`? At least, we should delete the commented-out code.